### DISCUSSION TITLE
希望可以有一个类似于opencv matchTemplate的功能

### DISCUSSION THREAD

**Discussion by Pycoder0901** [Meituan-Dianping/vision-ui#17] [General]
现在我看是已经有了目标检测的功能了 可以再优化一下

就是传一个小图标，然后返回这个小图标在大图中的位置坐标。我发现opencv matchTemplate有一些识别不到，但是vision-ui的模型倒是可以识别到 ，所以希望大佬可以优化下这个功能，万分感激！！！！

**Comment by BrightHai** [comment]
你好，有几个方案可以参考：
OpenCV
1. OpenCV Template match适用尺度不变的场景，小图标完整来自大图标
2. OpenCV SIFT/SURF匹配，计算量比较大也看小图标在大图中的区分性

目标识别
1. 识别到一个目标，用(x1,y1,x2,y2)计算点击坐标
2. 识别多个，先简单用顺序确定，同时我们也在探索在这个基础上建立定位坐标系

**Reply by Pycoder0901** [reply]
"识别到一个目标，用(x1,y1,x2,y2)计算点击坐标".目前不知道识别的每一个目标的 x1,y1,x2,y2怎么获取，打印dets都是类似于这种数据[8.86824768e+02 1.88550146e+03 9.63793945e+02 1.96115308e+03
  7.54326403e-01 2.00000000e+00]

**Reply by BrightHai** [reply]
boxes, scores, cls_inds = dets[:, :4], dets[:, 4], dets[:, 5]   在img_show有对boxes解析为xy的方法

**Reply by Pycoder0901** [reply]
OK了 非常感谢大佬

**Reply by Pycoder0901** [reply]
有一些没有识别到的图标 要怎么训练呢 我看ui_det_v1.onnx已经是打好包的了
